src/functions/tarsUpload/framework/adapter/fetch/StubBatchFetcher.ts

```diff
@@ -0,0 +1,38 @@
+import { IBatchFetcher } from '../../../application/secondary/IBatchFetcher';
+import { injectable, inject } from 'inversify';
+import { StandardCarTestCATBSchema } from '@dvsa/mes-test-schema/categories/B';
+import axios, { AxiosInstance } from 'axios';
+import * as zlib from 'zlib';
+import { ITestResultHTTPConfig } from '../upload/ITestResultHTTPConfig';
+import { TYPES } from '../../di/types';
+@injectable()
+export class StubBatchFetcher implements IBatchFetcher {
+
+  private endpoint: string;
+  axios: AxiosInstance;
+
+  private nextBatch: StandardCarTestCATBSchema[] = [];
+  constructor(
+    @inject(TYPES.TestResultHTTPConfig) private testResultHttpConfig: ITestResultHTTPConfig,
+  ) {
+    this.axios = axios.create({
+    });
+    this.endpoint = testResultHttpConfig.endpoint;
+  }
+
+  fetchNextUploadBatch(): Promise<StandardCarTestCATBSchema[]> {
+    return new Promise((resolve, reject) => {
+      const result = this.axios.get(this.endpoint);
+      result.then((response) => {
```

Comment:
If this intended to be a real implementation, it should probably have some error handling for the various things that can go wrong (request errors, not valid b64, not valid gzip, not valid JSON, etc).